src/error.rs

```diff
@@ -11,3 +14,9 @@ impl From<CurlError> for TealdeerError {
         TealdeerError::UpdateError(format!("Curl error: {}", err.to_string()))
     }
 }
+
+impl From<IoError> for TealdeerError {
+    fn from(err: IoError) -> TealdeerError {
+        TealdeerError::ConfigError(format!("Io error: {}", err))
+    }
+}
```

Comment:
Hm, I'm not sure if we should convert `IoError`s to `ConfigError`, since an `IoError` could also occur in cache handling code for example. Maybe it would be better to explicitly convert with `.map_err(|e| ConfigError(...))`?

Reply:
That's a lot of extra code, how about moving this impl to only be visible for the config module. Does that even work?

Reply:
I don't think that works. In the future, [catch](https://doc.rust-lang.org/unstable-book/language-features/catch-expr.html) will probably help in these cases.

If you want we can leave the impl in there for now. But I'm a bit uneasy about it, since when we add more I/O related code to cache handling or something like that, an error might wrongly be reported as a `ConfigError` when forgetting to properly convert the error.